polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -275,6 +274,89 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// The overarching event type.
+		#[pallet::no_default_bounds]
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The runtime `Origin` type.
+		#[pallet::no_default_bounds]
+		type RuntimeOrigin: From<Origin> + From<<Self as SysConfig>::RuntimeOrigin>;
+
+		/// The runtime `Call` type.
+		#[pallet::no_default_bounds]
+		type RuntimeCall: Parameter
+			+ GetDispatchInfo
+			+ Dispatchable<
+				RuntimeOrigin = <Self as Config>::RuntimeOrigin,
+				PostInfo = PostDispatchInfo,
+			>;
+	}
+
+	/// Default implementations of [`DefaultConfig`], which can be used to implement [`Config`]
+	pub mod config_preludes {
+		use super::*;
+		use frame_support::{
+			derive_impl, register_default_impl,
+			traits::{Everything, ConstU32, Equals},
+		};
+		use xcm_builder::{
+			IsConcrete,
+			LocationWithAssetFilters,
+		};
+
+		pub struct TestDefaultConfig;
```

Comment:
I belive we should be able to make `struct TestDefaultConfig` be `struct TestDefaultConfig<Some, Further, Parameterization>` if you are unsure about any of the things hardcoded in the following `parameter_types!`.

Reply:
That would be extremely helpful, however, I'm getting an error when trying to specify these generics in the calling `derive_impl(TestDefaultConfig<Some, Further, Parametrization>)`. Is this supposed to work or it should be added to the macro?

Reply:
I don't think there is a fundamental blocker for it, but it is quite likely that the syntax is not supported. Please proceed with @gupnik.